--- quagga-0.98.3/zebra/zebra_rib.c~	2004-12-07 21:12:56.000000000 +0000
+++ quagga-0.98.3/zebra/zebra_rib.c	2005-06-24 22:17:05.230996344 +0000
@@ -396,6 +396,8 @@
 		      {
 			SET_FLAG (nexthop->flags, NEXTHOP_FLAG_RECURSIVE);
 			nexthop->rtype = newhop->type;
+			if (newhop->type == NEXTHOP_TYPE_BLACKHOLE)
+			  nexthop_blackhole_add (rib);
 			if (newhop->type == NEXTHOP_TYPE_IPV4 ||
 			    newhop->type == NEXTHOP_TYPE_IPV4_IFINDEX)
 			  nexthop->rgate.ipv4 = newhop->gate.ipv4;
