--- quagga-0.98.3/zebra/zebra_rib.c~	2009-12-21 11:40:24.000000000 +0100
+++ quagga-0.98.3/zebra/zebra_rib.c	2010-03-23 00:00:00.000000000 +0100
@@ -384,6 +384,8 @@ nexthop_active_ipv4 (struct rib *rib, st
 		      {
 			SET_FLAG (nexthop->flags, NEXTHOP_FLAG_RECURSIVE);
 			nexthop->rtype = newhop->type;
+			if (newhop->type == NEXTHOP_TYPE_BLACKHOLE)
+			    nexthop_blackhole_add (rib);
 			if (newhop->type == NEXTHOP_TYPE_IPV4 ||
 			    newhop->type == NEXTHOP_TYPE_IPV4_IFINDEX)
 			  nexthop->rgate.ipv4 = newhop->gate.ipv4;
