--- quagga-0.99.22.4/configure.ac	2013-09-04 21:25:43.000000000 +0200
+++ quagga-0.99.22.4.new/configure.ac	2014-05-02 18:39:27.281182113 +0200
@@ -1050,45 +1050,15 @@
 dnl -----------------------
 dnl check proc file system.
 dnl -----------------------
-if test "$netlink" != yes; then
-  if test -r /proc/net/dev; then
     AC_DEFINE(HAVE_PROC_NET_DEV,,/proc/net/dev)
     IF_PROC=if_proc.o
-  fi
-  if test -r /proc/net/if_inet6; then
     AC_DEFINE(HAVE_PROC_NET_IF_INET6,,/proc/net/if_inet6)
-    IF_PROC=if_proc.o
-  fi
-fi
 AC_SUBST(IF_PROC)
 
 dnl -----------------------------
 dnl check ipforward detect method
 dnl -----------------------------
-AC_CACHE_CHECK([ipforward method], [quagga_cv_ipforward_method],
-[if test x$cross_compiling = xyes; then
-  if test x"$opsys" = x"gnu-linux"; then
-    quagga_cv_ipforward_method=/proc/net/snmp
-  else
-    quagga_cv_ipforward_method=/dev/ip
-  fi
-else
-  for quagga_cv_ipforward_method in /proc/net/snmp /dev/ip /dev/null;
-  do
-    test x`ls $quagga_cv_ipforward_method 2>/dev/null` = x"$quagga_cv_ipforward_method" && break
-  done
-fi
-case $quagga_cv_ipforward_method in
-  "/proc/net/snmp")  quagga_cv_ipforward_method="proc";;
-  "/dev/ip")         
-                     case "$host" in
-                       *-nec-sysv4*)  quagga_cv_ipforward_method="ews";;
-                       *-freebsd*)    quagga_cv_ipforward_method="sysctl";;
-                       *)             quagga_cv_ipforward_method="solaris";;
-                     esac;;
-  *)                 quagga_cv_ipforward_method="sysctl";;
-esac])
-IPFORWARD=ipforward_${quagga_cv_ipforward_method}.o
+IPFORWARD=ipforward_proc.o
 AC_SUBST(IPFORWARD)
 
 AC_CHECK_FUNCS(getaddrinfo, [have_getaddrinfo=yes], [have_getaddrinfo=no])
