diff -urNp -x '*.orig' quagga-0.99.23.org/vtysh/vtysh.c quagga-0.99.23/vtysh/vtysh.c
--- quagga-0.99.23.org/vtysh/vtysh.c	2014-06-24 07:14:20.000000000 +0200
+++ quagga-0.99.23/vtysh/vtysh.c	2023-02-02 08:51:40.358659811 +0100
@@ -682,6 +682,7 @@ new_completion (char *text, int start, i
         rl_completion_append_character = '\0';
     }
 
+  rl_completion_suppress_append = 1;
   return matches;
 }
 
