diff -urN quagga-0.99.11/vtysh/vtysh_main.c.orig quagga-0.99.11/vtysh/vtysh_main.c
--- quagga-0.99.11/vtysh/vtysh_main.c.orig	2008-08-14 16:43:42.000000000 +0200
+++ quagga-0.99.11/vtysh/vtysh_main.c	2009-04-10 01:53:47.000000000 +0200
@@ -42,6 +42,7 @@
 
 /* Configuration file name and directory. */
 char config_default[] = SYSCONFDIR VTYSH_DEFAULT_CONFIG;
+char history_file[MAXPATHLEN];
 
 /* Flag for indicate executing child command. */
 int execute_flag = 0;
@@ -184,8 +185,10 @@
     {
       using_history();
       last = previous_history();
-      if (!last || strcmp (last->line, line_read) != 0)
+      if (!last || strcmp (last->line, line_read) != 0) {
 	add_history (line_read);
+	append_history(1,history_file);
+      }
     }
      
   return (line_read);
@@ -348,10 +351,13 @@
   sigsetjmp (jmpbuf, 1);
   jmpflag = 1;
 
+  snprintf(history_file, sizeof(history_file), "%s/.history_quagga", getenv("HOME"));
+  read_history(history_file);
   /* Main command loop. */
   while (vtysh_rl_gets ())
     vtysh_execute (line_read);
 
+  history_truncate_file(history_file,1000);
   printf ("\n");
 
   /* Rest in peace. */
