--- quagga-0.99.13/configure.ac.orig	2009-07-11 19:57:36.000000000 +0200
+++ quagga-0.99.13/configure.ac	2009-07-11 20:03:30.000000000 +0200
@@ -951,37 +951,15 @@
 dnl -----------------------
 dnl check proc file system.
 dnl -----------------------
-if test "$netlink" != yes; then
-  if test -r /proc/net/dev; then
     AC_DEFINE(HAVE_PROC_NET_DEV,,/proc/net/dev)
     IF_PROC=if_proc.o
-  fi
-  if test -r /proc/net/if_inet6; then
     AC_DEFINE(HAVE_PROC_NET_IF_INET6,,/proc/net/if_inet6)
-    IF_PROC=if_proc.o
-  fi
-fi
 AC_SUBST(IF_PROC)
 
 dnl -----------------------------
 dnl check ipforward detect method
 dnl -----------------------------
-AC_CACHE_CHECK([ipforward method], [quagga_cv_ipforward_method],
-[for quagga_cv_ipforward_method in /proc/net/snmp /dev/ip /dev/null;
-do
-  test x`ls $quagga_cv_ipforward_method 2>/dev/null` = x"$quagga_cv_ipforward_method" && break
-done
-case $quagga_cv_ipforward_method in
-  "/proc/net/snmp")  quagga_cv_ipforward_method="proc";;
-  "/dev/ip")         
-                     case "$host" in
-                       *-nec-sysv4*)  quagga_cv_ipforward_method="ews";;
-                       *-freebsd*)    quagga_cv_ipforward_method="sysctl";;
-                       *)             quagga_cv_ipforward_method="solaris";;
-                     esac;;
-  *)                 quagga_cv_ipforward_method="sysctl";;
-esac])
-IPFORWARD=ipforward_${quagga_cv_ipforward_method}.o
+IPFORWARD=ipforward_proc.o
 AC_SUBST(IPFORWARD)
 
 AC_CHECK_FUNCS(getaddrinfo, [have_getaddrinfo=yes], [have_getaddrinfo=no])
